app/src/experiments.py

```diff
@@ -0,0 +1,153 @@
+
+
+pipeline_wrapper = PipelineWrapper()
+pipeline_wrapper.setup()
+
+
+def get_answer(example: dict) -> TaskOutput:
+    question_json_str = example["input"][INPUT_COLUMN_NAME]
+    question_obj = json.loads(question_json_str)
+    question = question_obj["caseworker_input"]
+
+    logger.info("Getting answer for: %r", question)
+    response = pipeline_wrapper.run_api(query=question)
+    assert len(response["llm"]["replies"]) == 1, pformat(response)
+    return response["llm"]["replies"][0].to_dict()
+
+
+def run(dataset: Dataset, client: Client) -> None:
+    run_experiment(
+        dataset=dataset,
+        task=get_answer,
+        evaluators=[precision, accuracy],
+        client=client,
+    )
+
+
+INPUT_COLUMN_NAME = "Input"
+OUTPUT_COLUMN_NAME = "Output"
+
+
+def export_dataset(dataset_name: str, filename: str, client: Client) -> None:
+    "Retrieve a dataset by name and save it as a CSV file. Useful for exporting a dataset from the deployed Phoenix."
+    try:
+        dataset = client.datasets.get_dataset(dataset=dataset_name)
+        # Convert the dataset to a pandas DataFrame to save as CSV
+        df = dataset.to_dataframe()
+
+        # Remove the extraneous 'input' and 'output' keys and just keep the values
+        # This ensures that when the CSV file is imported, the input and output columns are the same as in the deployed Phoenix
+        df[INPUT_COLUMN_NAME] = df["input"].apply(lambda x: x[INPUT_COLUMN_NAME])
+        df[OUTPUT_COLUMN_NAME] = df["output"].apply(lambda x: x[OUTPUT_COLUMN_NAME])
+        # Convert metadata to JSON string so it can be correctly parsed when running experiments if needed
+        df["metadata"] = df["metadata"].apply(lambda x: json.dumps(x))
+        df = df[[INPUT_COLUMN_NAME, OUTPUT_COLUMN_NAME, "metadata"]]
+        df.to_csv(filename, index=False)
+    except ValueError as e:
+        logger.error("Error retrieving dataset: %s", e)
+        print_datasets(client)
+
+
+def print_datasets(client: Client) -> list:
+    all_datasets = client.datasets.list()
+    logger.info("%d available datasets:\n%s", len(all_datasets), pformat(all_datasets))
+    return all_datasets
+
+
+def import_dataset(client: Client, filename: str, dataset_name: str) -> Any:
+    """
+    Import a dataset from a CSV file into Phoenix.
+    Useful for loading dataset locally to test experiments.
+    Reminder to reduce the number of rows in the CSV file for development.
+    """
+    return client.datasets.create_dataset(
+        name=dataset_name,
+        csv_file_path=filename,
+        input_keys=[INPUT_COLUMN_NAME],
+        output_keys=[OUTPUT_COLUMN_NAME],
+        # Caution: Note that the metadata column is different than on the deployed Phoenix
+        # In the Phoenix UI, click on each example to compare -- don't compare in the table view,
+        # which shows multiple examples and hides the "Input" and "Output" keys
+        metadata_keys=["metadata"],
```

Comment:
We don't use the `metadata` column currently, so any issues here have no effect but I left a code comment for future reference.